### PR TITLE
GHA: Use builtin docker

### PR DESCRIPTION
**Pull request by bastelfreak**

### CODE REVIEW

**Comment by bastelfreak** [comment]
@trevor-vaughan do you know if we install docker/podman on purpose? GitHub now provides those so I don't see a benefit in installing them explicitly.

**Comment by bastelfreak** [comment]
edit: If we should test with explicit installed docker I suggest we duplicate the CI jobs to test with both docker versions, explicitly installed and github-provided, because thats what we use for module testing.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -72,16 +72,6 @@ jobs:
         run: |
           gem install bundler -v '~> 1.17.3'
           bundle update
```
Comment (ekohl) [inline .github/workflows/test.yml:74]: We run `bundler-cache`, so we can drop this bit I think. And also upgrade this test to Ruby 3.2

**Review by ekohl — APPROVED** [review]
Could use a rebase now, but I think this should be good to be merged.

**Pull request merged**